test/integration.js

```diff
@@ -49,4 +57,52 @@ describe('integrations', function() {
       vfs.dest(outputBase),
     ], done);
   });
+
+  it('(*nix) sources a directory, creates a symlink and copies the symlink', function(done) {
+    if (isWindows) {
+      this.skip();
+      return;
+    }
+
+    function assert(files) {
+      var symlinkResult = fs.readlinkSync(outputSymlink);
+      var destResult = fs.readlinkSync(outputDirpathSymlink);
+
+      expect(symlinkResult).toEqual(inputDirpath);
+      expect(destResult).toEqual(inputDirpath);
+      expect(files[0].symlink).toEqual(inputDirpath);
+    }
+
+    pipe([
+      vfs.src(inputDirpath),
+      vfs.symlink(symlinkDirpath),
+      vfs.dest(outputDirpath),
+      concat(assert),
+    ], done);
+  });
+
+  it('(windows) sources a directory, creates a junction and copies the junction', function(done) {
+    if (!isWindows) {
+      this.skip();
+      return;
+    }
+
+    function assert(files) {
+      // Junctions add an ending separator
+      var expected = inputDirpath + path.sep;
+      var symlinkResult = fs.readlinkSync(outputSymlink);
+      var destResult = fs.readlinkSync(outputDirpathSymlink);
+
+      expect(symlinkResult).toEqual(expected);
+      expect(destResult).toEqual(expected);
+      expect(files[0].symlink).toEqual(inputDirpath);
+    }
+
+    pipe([
+      vfs.src(inputDirpath),
+      vfs.symlink(symlinkDirpath),
+      vfs.dest(outputDirpath),
+      concat(assert),
+    ], done);
+  });
```

Comment:
I was thinking it might make sense to additionally test the case of directly sourcing a symlinked directory and copying it:

```js
it('(*nix) sources a symlink and copies it', function(done) {
    if (isWindows) {
      this.skip();
      return;
    }

    fs.mkdirSync(base);
    fs.mkdirSync(symlinkDirpath);
    fs.symlinkSync(inputDirpath, outputSymlink);

    function assert(files) {
      var destResult = fs.readlinkSync(outputDirpathSymlink);

      expect(destResult).toEqual(inputDirpath);
      expect(files[0].symlink).toEqual(inputDirpath);
    }

    pipe([
      vfs.src(outputSymlink, { resolveSymlinks: false }),
      vfs.dest(outputDirpath),
      concat(assert),
    ], done);
  });

  it('(windows) sources a junction and copies it', function(done) {
    if (!isWindows) {
      this.skip();
      return;
    }

    fs.mkdirSync(base);
    fs.mkdirSync(symlinkDirpath);
    fs.symlinkSync(inputDirpath, outputSymlink, 'junction');

    function assert(files) {
      // Junctions add an ending separator
      var expected = inputDirpath + path.sep;
      var destResult = fs.readlinkSync(outputDirpathSymlink);

      expect(destResult).toEqual(expected);
      expect(files[0].symlink).toEqual(inputDirpath);
    }

    pipe([
      vfs.src(outputSymlink, { resolveSymlinks: false }),
      vfs.dest(outputDirpath),
      concat(assert),
    ], done);
  });
```

Reply:
Added in 9c021e4